OpenSim/Simulation/Model/ContactMesh.cpp

```diff
@@ -145,7 +148,7 @@ void ContactMesh::generateDecorations(bool fixed, const ModelDisplayHints& hints
         const auto& X_BF = getFrame().findTransformInBaseFrame();
         const auto& X_FP = getTransform();
         const auto X_BP = X_BF * X_FP;
-        geometry.push_back(SimTK::DecorativeMeshFile(get_filename())
+        geometry.push_back(SimTK::DecorativeMesh(*_decorativeGeometry)
```

Comment:
Agreed, however I think that needs to be handled somewhere else so that the component is not processed during traversal, otherwise each component will deal with it in an ad-hoc way.